### PR TITLE
Fix godoc display issues, other doc fixes

### PR DESCRIPTION
**Pull request by c633**

### CODE REVIEW

**Comment by masomel** [comment]
@arlolra @Liamsi Should we add documentation to the executables' `main` package? Original discussion: https://github.com/coniks-sys/coniks-go/pull/131#discussion_r89150645

**Comment by c633** [comment]
@masomel : I rebased this branch with `master` and moved some subpackages to `/internal` in bdcf27b6fe1674eee244bd0c75b8b103aaba3cb4. What do you think? godoc will not show `bar` package if it is an internal package of `foo` (`foo/internal/bar`).

You can use `git reset --soft origin/fix-doc` in your local branch to avoid conflict.

**Comment by Liamsi** [comment]
> @arlolra @Liamsi Should we add documentation to the executables' main package? Original discussion: #131 (comment)

I agree with @c633 that this isn't really necessary.

**Comment by c633** [comment]
Any further comments on this?

**Comment by masomel** [comment]
> Any further comments on this?

Let me do one more pass over this.

File: merkletree/proof.go [hunk 1]
```diff
@@ -114,6 +114,11 @@ func (ap *AuthenticationPath) Verify(key, value, treeHash []byte) bool {
 	return true
 }
 
+// ProofType returns the type of ap. It does a comparison
+// between the leaf index and the lookup index to determine
+// the proof type in the first time when this is called.
+// After that, it can be called many times without
+// rerunning the comparison.
```
Comment (arlolra) [inline merkletree/proof.go]: The term for this is "memoized".

**Comment by arlolra** [comment]
Is this pull meant to still be in WIP?  If so, what's left to do?  It seems pretty reasonable to me.

**Comment by masomel** [comment]
>  Is this pull meant to still be in WIP? If so, what's left to do?

I'm doing one last pass over all the documentation to make sure we didn't miss anything. I don't think it needs to be in WIP for that though.

**Comment by masomel** [comment]
So, I know that we had talked about not needing summaries for executable commands, but their main packages still appear in godoc, and I think it's awkward that these packages don't have a summary. What if the summary just said something like "Executable X. See README for usage information"?

**Comment by arlolra** [comment]
Sounds reasonable.

**Comment by arlolra** [comment]
FYI, I fixed up the go formatting issues and rebased.

**Comment by c633** [comment]
LGTM. Thank you @marcela @arlolra

**Comment by masomel** [comment]
Is this ready to be merged?

**Comment by c633** [comment]
From my side, yes.

**Pull request closed without merging**